modules/custom/openy_campaign/src/Plugin/Block/CampaignUserMenuBlock.php

```diff
@@ -63,7 +63,8 @@ public static function create(ContainerInterface $container, array $configuratio
   public function build() {
     $build = [];
 
-    // Disable block cache.
+    // The block is rendered for each user separately.
+    // We can't cache it.
```

Comment:
we have user cache tags, or user context. https://www.drupal.org/docs/8/api/cache-api/cache-contexts
Let's work on this after release.